src/engine/source/hlp/src/SpecificParsers.cpp

```diff
@@ -390,116 +468,254 @@ bool parseURL(const char **it, char endToken, URLResult &result) {
         }
-        else {
-            result.top_level_domain = labels.back() + "." + result.top_level_domain;
+        else
+        {
+            result.top_level_domain =
+                labels.back() + "." + result.top_level_domain;
         }
         labels.pop_back();
     }
 
     // Registered domain
-    if (result.top_level_domain.empty()) {
+    if(result.top_level_domain.empty())
+    {
         result.registered_domain = labels.back();
     }
-    else {
-        result.registered_domain = labels.back() + "." + result.top_level_domain;
+    else
+    {
+        result.registered_domain =
+            labels.back() + "." + result.top_level_domain;
     }
     labels.pop_back();
 
     // Subdomain
-    for (auto label : labels) {
-        if (result.subdomain.empty()) {
+    for(auto label : labels)
+    {
+        if(result.subdomain.empty())
+        {
             result.subdomain = label;
         }
-        else {
+        else
+        {
             result.subdomain = result.subdomain + "." + label;
         }
     }
 
     // Address
     result.address = result.domain;
 
-    // Validate if all fields are complete to identify a Fully Qualified Domain Name
-    if (validate_FQDN) {
-        if (result.top_level_domain.empty()) {
-            //TODO log error
+    // Validate if all fields are complete to identify a Fully Qualified Domain
+    // Name
+    if(validate_FQDN)
+    {
+        if(result.top_level_domain.empty())
+        {
+            // TODO log error
             return false;
         }
-        if (result.registered_domain.empty()) {
-            //TODO log error. One for each missing field?
+        if(result.registered_domain.empty())
+        {
+            // TODO log error. One for each missing field?
             return false;
         }
-        if (result.subdomain.empty()) {
-            //TODO log error. One for each missing field?
+        if(result.subdomain.empty())
+        {
+            // TODO log error. One for each missing field?
             return false;
         }
     }
 
     return true;
 }
+
+enum class UAState
+{
+    Product,
+    Comment,
+};
+
+bool parseUserAgent(const char **it, char endToken, UserAgentResult &ret)
+{
+    const char *start = *it;
+    const char *ev = *it;
+
+    // NOTE: This will try to validate 'some' part of the user-agent standard as
+    // is defined in https://datatracker.ietf.org/doc/html/rfc7231#section-5.5.3
+    // it will accept as valid only user agents with the form of - token/token
+    // (comment) token/token - or - token/token token/token - Other ~valid~
+    // formats of user agents like:
+    //     - Mozilla/5.0 (Windows NT 6.1; WOW64; Trident/7.0; rv:11.0) like Gecko
+    //     - product1/version1 product2/version2 product3
+    //     - product/version (comment, (nested comment))
+    // are not considered valid because it would be impossible to differentiate
+    // from a 'literal'
+
+    bool done = false;
```

Comment:
We should follow the same format as implemented in other parts of the code, like line 177